reframe/frontend/dependency.py

```diff
@@ -167,3 +167,21 @@ def visit(node, path):
             cases_by_name[c.check.name] = [c]
 
     return list(itertools.chain(*(cases_by_name[n] for n in visited)))
+
+
+def create_deps_count(graph):
+
+    # TODO: create unit test
+    deps_count = {}
+    for test, deps in graph.items():
+        try:
```

Comment:
No, this I understand. My point here was about whether the double `try/except` clause is actually needed.

Reply:
Also I don't understand the first `+= 1`? Sth like the following should be enough I guess:

```python
for test, deps in graph.items():
    ref_count.setdefault(test, 0)
    for d in deps:
        try:
            ref_count[test] += 1
        except KeyError:
            ref_count[test] = 0
```

Reply:
Yes, you are right, I don't need the +1.
By the way, I see that you are using the ``test`` variable as key in the inner loop instead of using ``dep``, but currently the content of ``ref_count[c]``, for a given check c, is the number of checks depending on c. Should I change the ref_count data meaning?